addons/mrp/models/mrp_workorder.py

```diff
@@ -421,18 +431,40 @@ def record_production(self):
         # Update workorder quantity produced
         self.qty_produced += self.qty_producing
 
+        rounding = self.production_id.product_uom_id.rounding
+        previous_wo = self.env['mrp.workorder'].search([
+            ('next_work_order_id', '=', self.id)])
         # Suggest a finished lot on the next workorder
-        if self.next_work_order_id and self.product_tracking != 'none' and (not self.next_work_order_id.finished_lot_id or self.next_work_order_id.finished_lot_id == self.finished_lot_id):
-            self.next_work_order_id._defaults_from_finished_workorder_line(self.finished_workorder_line_ids)
+        if self.next_work_order_id and self.product_tracking != 'none' and (not self.next_work_order_id.finished_lot_id or self.next_work_order_id.finished_lot_id == self.finished_lot_id) and self.next_work_order_id.state != 'done':
```

Comment:
We could probably remove the if now that quantity are taken into account. And do all the job in _defaults_from_finished_workorder_line

Reply:
method _defaults_from_finished_workorder_line is responsible for assigning the lot id if exist in current wo and returns the True or False if candidate is found or not.
And this code is -
1) To suggest the previously produced quantity even if lot id is not present.
So it this case we can not say that candidate is found but need to suggest the quantity thats why change the code here.